cpu/stm32/periph/eth.c

```diff
@@ -86,6 +97,10 @@ static char rx_buffer[ETH_RX_DESCRIPTOR_COUNT][ETH_RX_BUFFER_SIZE];
 /* Netdev used in RIOT's API to upper layer */
 netdev_t *_netdev;
 
+/* Used for checking the link status */
+static uint8_t _link_status = 0;
+static uint8_t _first_call = 1;
```

Comment:
No need to have two variables for this. It is a simple state machine with theses states:

```
+------------------+  1  +---------------+  2  +-------------+
| Link down (init) | --> |    Link up    | --> | Notified Up |
+------------------+     +---------------+     +-------------+
                                 ^                    |
                                 | 1                  | 3
                                 |                    v
                         +---------------+  4  +-------------+
                         | Notified Down | <-- |  Link Down  |
                         +---------------+     +-------------+
```

And these events triggering the transitions:

1. Link goes up
2. Upper layer gets informed that link is up
3. link goes down
4. Upper layer gets informed that link is down

Tracking the current state fits nicely into a single byte.

Reply:
No need to distinguish between the initial link down state and the notified down state